src/cartridges/int_adyen_overlay/cartridge/scripts/adyenCheckout.js

```diff
@@ -183,11 +184,29 @@ function doPaymentsCall(order, paymentInstrument, paymentRequest) {
         : null;
     }
 
+    const challengeResultCodes = [
+      constants.RESULTCODES.IDENTIFYSHOPPER,
+      constants.RESULTCODES.CHALLENGESHOPPER
+    ];
+
+    const acceptedResultCodes = [
+      constants.RESULTCODES.AUTHORISED,
+      constants.RESULTCODES.PENDING,
+      constants.RESULTCODES.RECEIVED,
+      constants.RESULTCODES.REDIRECTSHOPPER
+    ];
+
+    const presentToShopperResultCodes = [constants.RESULTCODES.PRESENTTOSHOPPER];
+
+    const refusedResultCodes = [
+      constants.RESULTCODES.CANCELLED,
+      constants.RESULTCODES.ERROR,
+      constants.RESULTCODES.REFUSED
+    ];
+
+    const {resultCode} = paymentResponse;
     // Check the response object from /payment call
-    if (
-      paymentResponse.resultCode === 'IdentifyShopper' ||
-      paymentResponse.resultCode === 'ChallengeShopper'
-    ) {
+    if (challengeResultCodes.indexOf(resultCode) !== -1) {
```

Comment:
I would rename this to ThreeDS2ResultCodes for example as this is Identify and Challenge Shopper